File: icpc1.js
//console.log("Hello world "); 


const batatro = (cards) => {

    let dp = Array(cards.length + 1).fill(-Infinity); 
    dp[0] = 0;  // base case ya basha! 

    for (let i = 0; i < cards.length; i++) {
        const [type, value] = cards[i]; 


        for (let size = cards.length; size > 0; size--) {
            if (dp[size - 1] !== -Infinity) {
                if (type === 'add') {
                    dp[size] = Math.max(dp[size], dp[size - 1] + value);
                } else if (type === 'multiply') {
                    dp[size] = Math.max(dp[size], dp[size - 1] * value);
                }
            }
        }
    }

    return dp.slice(1); // return max scores for subsequences of size
}

File: icpc1.py
def balatro(cards):

    dp = [-float('inf')] * (len(cards) + 1)
    dp[0] = 0  # base case

    for i in range(len(cards)):
        card_type, value = cards[i]

        for size in range(len(cards), 0, -1):
            if dp[size - 1] != -float('inf'):
                if card_type == 'a':
                    dp[size] = max(dp[size], dp[size - 1] + value)
                elif card_type == 'm':

                    dp[size] = max(dp[size], dp[size - 1] * value)

    return dp[1:]  


